src/main/java/ch/njol/skript/effects/EffOperations.java

```diff
@@ -0,0 +1,209 @@
+import ch.njol.skript.lang.SyntaxStringBuilder;
+import ch.njol.skript.registrations.Classes;
+import ch.njol.skript.util.LiteralUtils;
+import ch.njol.skript.util.Patterns;
+import ch.njol.skript.util.Utils;
+import ch.njol.util.Kleenean;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.Nullable;
+import org.skriptlang.skript.lang.arithmetic.Arithmetics;
+import org.skriptlang.skript.lang.arithmetic.Operation;
+import org.skriptlang.skript.lang.arithmetic.OperationInfo;
+import org.skriptlang.skript.lang.arithmetic.Operator;
+import org.skriptlang.skript.log.runtime.SyntaxRuntimeErrorProducer;
+
+import java.util.function.Function;
+
+@Name("Operations")
+@Description("Perform multiplication, division, or exponentiation operations on variable objects "
+	+ "(i.e. numbers, vectors, timespans, and other objects from addons). "
+	+ "Literals cannot be used on the left-hand side.")
+@Example("""
+	set {_num} to 1
+	multiply {_num} by 10
+	divide {_num} by 5
+	raise {_num} to the power of 2
+	""")
+@Example("""
+	set {_nums::*} to 15, 21 and 30
+	divide {_nums::*} by 3
+	multiply {_nums::*} by 5
+	raise {_nums::*} to the power of 3
+	""")
+@Example("""
+	set {_vector} to vector(1,1,1)
+	multiply {_vector} by vector(4,8,16)
+	divide {_vector} by 2
+	""")
+@Example("""
+	set {_timespan} to 1 hour
+	multiply {_timespan} by 3
+	""")
+@Example("""
+	# Will error due to literal
+	multiply 1 by 2
+	divide 10 by {_num}
+	""")
+@Since("INSERT VERSION")
+public class EffOperations extends Effect implements SyntaxRuntimeErrorProducer {
+
+	private static final Patterns<Operator> PATTERNS = new Patterns<>(new Object[][]{
+		{"multiply %~objects% by %object%", Operator.MULTIPLICATION},
+		{"divide %~objects% by %object%", Operator.DIVISION},
+		{"raise %~objects% to [the] (power|exponent) [of] %object%", Operator.EXPONENTIATION}
+	});
+
+	static {
+		Skript.registerEffect(EffOperations.class, PATTERNS.getPatterns());
+	}
+
+	private Operator operator;
+	private Expression<?> base;
+	private Expression<?> operative;
+	private Node node;
+	private Operation<Object, Object, Object> operation = null;
+
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		operator = PATTERNS.getInfo(matchedPattern);
+		base = exprs[0];
+		Class<?>[] baseAccepted = base.acceptChange(ChangeMode.SET);
+		if (baseAccepted == null) {
+			Skript.error("This expression cannot be operated on.");
+			return false;
+		}
+		operative = LiteralUtils.defendExpression(exprs[1]);
+		node = getParser().getNode();
+		Class<?> operativeType = operative.getReturnType();
+		Class<?> baseType = base.getReturnType();
+		// Ensure 'baseType' is referencing a non-array class
+		if (baseType.isArray())
+			baseType = baseType.getComponentType();
+		if (operativeType.equals(Object.class) && baseType.equals(Object.class)) {
+			// Both are object, so we should do the checks within '#execute'
```

Comment:
we can still check `Arithmetics.getAllReturnTypes(operator);` against the accepted classes for SET